<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Step 0: Setup and Defaults | ADSNPheno Book</title>
  <meta name="description" content="Please note that this tutorial book explains how to use the ADSNPheno computational pipeline to better understand the role of non-coding SNPs in impacting gene regulation, associated with changes in biological pathways and phenotypes in different health conditions." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Step 0: Setup and Defaults | ADSNPheno Book" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Please note that this tutorial book explains how to use the ADSNPheno computational pipeline to better understand the role of non-coding SNPs in impacting gene regulation, associated with changes in biological pathways and phenotypes in different health conditions." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Step 0: Setup and Defaults | ADSNPheno Book" />
  
  <meta name="twitter:description" content="Please note that this tutorial book explains how to use the ADSNPheno computational pipeline to better understand the role of non-coding SNPs in impacting gene regulation, associated with changes in biological pathways and phenotypes in different health conditions." />
  

<meta name="author" content="Saniya Khullar, Daifeng Wang, Ph.D." />


<meta name="date" content="2021-05-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="methods.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ADSNPheno Book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Please Note:</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#data-and-code-location"><i class="fa fa-check"></i><b>1.1</b> Data and Code Location</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#hardware-requirements"><i class="fa fa-check"></i><b>1.2</b> Hardware Requirements</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#support"><i class="fa fa-check"></i><b>1.3</b> Support</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#demo"><i class="fa fa-check"></i><b>1.4</b> Demo</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html"><i class="fa fa-check"></i><b>3</b> Step 0: Setup and Defaults</a>
<ul>
<li class="chapter" data-level="3.1" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html#default-datasets"><i class="fa fa-check"></i><b>3.1</b> Default Datasets:</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html#entrez-mapping-information"><i class="fa fa-check"></i><b>3.1.1</b> 1. Entrez Mapping Information:</a></li>
<li class="chapter" data-level="3.1.2" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html#transcription-factor-information"><i class="fa fa-check"></i><b>3.1.2</b> 2. Transcription Factor Information:</a></li>
<li class="chapter" data-level="3.1.3" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html#additional-gene-regulatory-information"><i class="fa fa-check"></i><b>3.1.3</b> 3. Additional Gene Regulatory Information:</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html#adjusting-default-code"><i class="fa fa-check"></i><b>3.2</b> Adjusting Default Code:</a></li>
<li class="chapter" data-level="3.3" data-path="step-0-setup-and-defaults.html"><a href="step-0-setup-and-defaults.html#default-settings"><i class="fa fa-check"></i><b>3.3</b> Default Settings:</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a>
<ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>5.1</b> Example one</a></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>5.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ADSNPheno Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="step-0-setup-and-defaults" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> Step 0: Setup and Defaults</h1>
<p><strong>Please setup the appropriate workspace environment</strong></p>
<p>Please note that it is important to download our ADSNPheno code from <a href="https://github.com/daifengwanglab/adsnpheno">https://github.com/daifengwanglab/adsnpheno</a> onto your local computer or server, where you can access it. It is recommended that you work with the layout of our existing folders (with our data files and code).</p>
<p>Then, you make adjustments as needed in the ADSNPheno folder, such as:</p>
<ul>
<li><p>adding your respective data files to the appropriate folders (please go to data folder)</p></li>
<li><p>changing the parameters in the code based on your needs and requirements (by going to code &gt; userInputForEachStep)</p></li>
<li><p>adapting our code to further suit your specific needs</p></li>
</ul>
<p>This is to help you adapt on our code as well, better understand the tools involved, and be more user-friendly.</p>
<p>When you download our pipeline ADSNPheno, as a zip file(“ADSNPheno-master.zip”), it hopefully will come with the sub-folders shown above (such as the “code” and “data” folders). Please unzip the folder and then note the complete file path location of the unzipped folder, which will be the variable adsnphenoDirectory. For instance, on Saniya’s computer, this ADSNPheno folder is located here: “F://organizedAlzheimers//ADSNPheno//,” so adsnphenoDirectory = “F://organizedAlzheimers//ADSNPheno//.”</p>
<p align="center">
<img width="400" src="directory.PNG">
</p>
<div id="default-datasets" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Default Datasets:</h2>
<p>Please note that we have shared 3 data sources that we have used for our analysis, which we believe may be general-purpose and helpful references for you. These are also found in the dataDefaults folder (data <span class="math inline">\(\rightarrow\)</span> dataDefaults) and are information on:</p>
<ul>
<li><ol style="list-style-type: decimal">
<li>Entrez Mapping Information</li>
</ol></li>
<li><ol start="2" style="list-style-type: decimal">
<li>Transcription Factor Information</li>
</ol></li>
<li><ol start="3" style="list-style-type: decimal">
<li>Additional Gene Regulatory Information</li>
</ol></li>
</ul>
<p align="center">
<img width="500" src="dataDefaults.PNG">
</p>
<p>Please see these respective sections below for more information about these data files and on how to appropriately edit this information. For your convenience, these default data files have been saved as an RData object, which can be loaded with this command:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="step-0-setup-and-defaults.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;data/generalDataForPipeline.RData&quot;</span>)</span></code></pre></div>
<p>These data objects are then loaded, which are then used by the pipeline.</p>
<p align="center">
<img width="500" src="loadingGeneralData.PNG">
</p>
<div id="entrez-mapping-information" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> 1. Entrez Mapping Information:</h3>
<p>To assist you with your gene set, please note that we have tried to build an extensive resource on some of the Entrez IDs for various genes so that you can map many genes to Entrez IDs. Please beware that Excel may distort some gene names like SEPT7 and MARC1 to numeric date values by mistake (we have tried to troubleshoot for some of those situations). It is not exhaustive, so please check additional sources such as NCBI for any genes you cannot find and make adjustments to this default data source, as needed, if you find any discrepancies. And, please email Saniya at <a href="mailto:skhullar2@wisc.edu" class="email">skhullar2@wisc.edu</a>, so she can update the files as needed (thanks for your help!). For many genes that begin with “LOC,” the entrez ID is often the numeric part of the gene name. This mapping will be used as a lookup in the pipeline to help with merging relationships, from disparate sources, reliably!</p>
<p>Please note an example of some rows of our “data//dataDefaults//entrezMappingInfo//infoDFMappingsGeneSymbolAndID.csv”:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="step-0-setup-and-defaults.html#cb2-1" aria-hidden="true" tabindex="-1"></a>someRows <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">200</span><span class="sc">:</span><span class="dv">205</span>)<span class="co"># 10 different rows of the CSV as an example</span></span>
<span id="cb2-2"><a href="step-0-setup-and-defaults.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(infoDF) <span class="co"># 59,307 rows and 2 columns</span></span></code></pre></div>
<pre><code>## [1] 58687     2</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="step-0-setup-and-defaults.html#cb4-1" aria-hidden="true" tabindex="-1"></a>infoDF[someRows,]  <span class="co"># geneSymbolName mapping to respective entrezIDs. Please double-check.</span></span></code></pre></div>
<pre><code>##     geneSymbolName entrezID
## 1           15.212     3805
## 2            47.11     3802
## 3             3635   201626
## 4           Jul-60    11054
## 5           19-Feb    60343
## 204        ABHD16B   140701
## 205        ABHD17A    81926
## 206      ABHD17AP1   728917
## 207      ABHD17AP3   648359
## 208      ABHD17AP4   729495
## 209      ABHD17AP5    91219</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="step-0-setup-and-defaults.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#     geneSymbolName  entrezID</span></span>
<span id="cb6-2"><a href="step-0-setup-and-defaults.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 1           15.212      3805</span></span>
<span id="cb6-3"><a href="step-0-setup-and-defaults.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 2            47.11      3802</span></span>
<span id="cb6-4"><a href="step-0-setup-and-defaults.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 3             3635    201626</span></span>
<span id="cb6-5"><a href="step-0-setup-and-defaults.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 4           Jul-60     11054</span></span>
<span id="cb6-6"><a href="step-0-setup-and-defaults.html#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 5           19-Feb     60343</span></span>
<span id="cb6-7"><a href="step-0-setup-and-defaults.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 200        ABHD14B     84836</span></span>
<span id="cb6-8"><a href="step-0-setup-and-defaults.html#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 201         ABHD15    116236</span></span>
<span id="cb6-9"><a href="step-0-setup-and-defaults.html#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 202     ABHD15-AS1 104355133</span></span>
<span id="cb6-10"><a href="step-0-setup-and-defaults.html#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 203        ABHD16A      7920</span></span>
<span id="cb6-11"><a href="step-0-setup-and-defaults.html#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 204        ABHD16B    140701</span></span>
<span id="cb6-12"><a href="step-0-setup-and-defaults.html#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 205        ABHD17A     81926</span></span></code></pre></div>
</div>
<div id="transcription-factor-information" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> 2. Transcription Factor Information:</h3>
</div>
<div id="additional-gene-regulatory-information" class="section level3" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> 3. Additional Gene Regulatory Information:</h3>
<p>Transcription Factor (TF) to Target Gene (TG) relationships based on TRRUST 2.0, which may serve as an additional source of TF-TG interactions. Information is also provided on some relationships and if the TF activates or represses the TG.</p>
<p>Please note an example of some rows of our “data//dataDefaults//additionalGeneRegulatoryInfo//Trrust2.csv.” Here, CombinedName basically uses the gene symbol names and 2 pipes to represent the relationship: “TF || RegulatedGene.” Thus, the 1st row has “AATF” as the TF and “CDKN1A” as the RegulatedGene, so CombinedName will be “AATF || CDKN1A”</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="step-0-setup-and-defaults.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(trrust2DF) <span class="co"># 9,395 rows of unique TF-TG relationships and these 5 columns (at least) are needed.  </span></span></code></pre></div>
<pre><code>## [1] 9395    5</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="step-0-setup-and-defaults.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#head(trrust2DF) # the first 5 rows of the TRRUST2.0 data source.</span></span>
<span id="cb9-2"><a href="step-0-setup-and-defaults.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     TF RegulatedGene  Source                             Info   CombinedName</span></span>
<span id="cb9-3"><a href="step-0-setup-and-defaults.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 AATF        CDKN1A Trrust2    TRRUST2 (Regulation: Unknown) AATF || CDKN1A</span></span>
<span id="cb9-4"><a href="step-0-setup-and-defaults.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 AATF          KLK3 Trrust2    TRRUST2 (Regulation: Unknown)   AATF || KLK3</span></span>
<span id="cb9-5"><a href="step-0-setup-and-defaults.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 AATF           MYC Trrust2 TRRUST2 (Regulation: Activation)    AATF || MYC</span></span>
<span id="cb9-6"><a href="step-0-setup-and-defaults.html#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 4 AATF          TP53 Trrust2    TRRUST2 (Regulation: Unknown)   AATF || TP53</span></span>
<span id="cb9-7"><a href="step-0-setup-and-defaults.html#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 ABL1           BAX Trrust2 TRRUST2 (Regulation: Activation)    ABL1 || BAX</span></span>
<span id="cb9-8"><a href="step-0-setup-and-defaults.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 6 ABL1          BCL2 Trrust2 TRRUST2 (Regulation: Repression)   ABL1 || BCL2</span></span></code></pre></div>
</div>
</div>
<div id="adjusting-default-code" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Adjusting Default Code:</h2>
</div>
<div id="default-settings" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Default Settings:</h2>
<p>Below, please note the main parameters to adjust (and please see step0_loadingInDefaultDatasets.R for additional parameters that may be varied)</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Variable Input Needed by You</th>
<th>Description</th>
<th>Default Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>adsnphenoDirectory</td>
<td>The region of the body that your gene expression data is for. In our example, it is the Hippocampus Ca1 Region</td>
<td>adsnphenoDirectory = “F://organizedAlzheimers//ADSNPheno//”</td>
</tr>
<tr class="even">
<td>tfsUsed</td>
<td>The sources of information for the Transcription Factors (TFs). Here, we use TFs found in Lambert and/or Jaspar as our default.</td>
<td>tfsUsed = “LambertAndJaspar”</td>
</tr>
<tr class="odd">
<td>transcriptionFactorsFilePath</td>
<td>In the dataDefaults &gt; transcriptionFactorInfo directory, this is the name of the CSV file with the list of TFs you want to use. Here, our list is for Lambert and/or Jaspar TFs.</td>
<td>transcriptionFactorsFilePath = “LambertAndJasparTFs.csv”</td>
</tr>
<tr class="even">
<td>geneToEntrezIDMappingPath</td>
<td>In the dataDefaults &gt; entrezMappingInfo directory, this is the name of the CSV file with 2 columns: [Symbols, entrezID]. Here, “Symbols” is the geneName (or # or numeric date), and “entrezID” is what we typically found as the entrezID.</td>
<td>geneToEntrezIDMappingPath = “infoDFMappingsGeneSymbolAndID.csv”</td>
</tr>
<tr class="odd">
<td>additionalGeneRegulatoryNetworkFilePath</td>
<td>In the dataDefaults <span class="math inline">\(\rightarrow\)</span> additionalGeneRegulatoryInfo directory, this is the name of the CSV file with external gold standard gene regulatory network information you would like to include in the final analysis. Here, we used TRRUST2.0. Our data file needs to contain at least these 5 columns: [TF, RegulatedGene, Source, Info, CombinedName]. CombinedName is basically a column based on the gene symbols and is separated with 2 pipes</td>
<td>additionalGeneRegulatoryNetworkFilePath = “Trrust2.csv”</td>
</tr>
<tr class="even">
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="odd">
<td>numberOfRoundingDigits</td>
<td>Please note that this is the # of digits after the decimal place for rounding any calculations performed during ADSNPheno.</td>
<td>numberOfRoundingDigits = 3</td>
</tr>
<tr class="even">
<td>includeTimeStamp</td>
<td>Please note that if this Boolean variable is set to TRUE, then additional time-stamped data files will be saved as the pipeline is run.</td>
<td>includeTimeStamp = TRUE</td>
</tr>
<tr class="odd">
<td>disease</td>
<td>Please note that this is where you would mention the disease or condition or relevant biological context. Here, we are analyzing the potential role of non-coding SNPs in Alzheimer’s disease, so we set disease = “Alzheimers.”</td>
<td>disease = “Alzheimers”</td>
</tr>
<tr class="even">
<td>outputPath</td>
<td>Please note that this is where the outputs of ADSNPheno would be stored. By default, ADSNPheno will build an output folder and store the corresponding datafiles and results in sub-folders there.</td>
<td>outputPath = “ADSNPheno//ADSNPhenoOutputs”</td>
</tr>
</tbody>
</table>
<p>We also have shared 3 data sources that we have used for our analysis, which we believe may be general-purpose and helpful references for you. These are also found in the dataDefaults folder (data <span class="math inline">\(\rightarrow\)</span> dataDefaults) and are information on:</p>
<ul>
<li><ol style="list-style-type: decimal">
<li>Additional Gene Regulatory Information: Transcription Factor (TF) to Target Gene (TG) relationships based on TRRUST 2.0, which may serve as an additional source of TF-TG interactions. Information is also provided on some relationships and if the TF activates or represses the TG.</li>
</ol></li>
</ul>
<p>Please note an example of some rows of our “data//dataDefaults//additionalGeneRegulatoryInfo//Trrust2.csv.” Here, CombinedName basically uses the gene symbol names and 2 pipes to represent the relationship: “TF || RegulatedGene.” Thus, the 1st row has “AATF”" as the TF and “CDKN1A”" as the RegulatedGene, so CombinedName will be “AATF || CDKN1A”</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="step-0-setup-and-defaults.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(trrust2DF) # 9,395 rows of unique TF-TG relationships and these 5 columns (at least) are needed.  </span></span>
<span id="cb10-2"><a href="step-0-setup-and-defaults.html#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="step-0-setup-and-defaults.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#head(trrust2DF) # the first 5 rows of the TRRUST2.0 data source.</span></span>
<span id="cb10-4"><a href="step-0-setup-and-defaults.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     TF RegulatedGene  Source                             Info   CombinedName</span></span>
<span id="cb10-5"><a href="step-0-setup-and-defaults.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 AATF        CDKN1A Trrust2    TRRUST2 (Regulation: Unknown) AATF || CDKN1A</span></span>
<span id="cb10-6"><a href="step-0-setup-and-defaults.html#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 AATF          KLK3 Trrust2    TRRUST2 (Regulation: Unknown)   AATF || KLK3</span></span>
<span id="cb10-7"><a href="step-0-setup-and-defaults.html#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 AATF           MYC Trrust2 TRRUST2 (Regulation: Activation)    AATF || MYC</span></span>
<span id="cb10-8"><a href="step-0-setup-and-defaults.html#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 4 AATF          TP53 Trrust2    TRRUST2 (Regulation: Unknown)   AATF || TP53</span></span>
<span id="cb10-9"><a href="step-0-setup-and-defaults.html#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 ABL1           BAX Trrust2 TRRUST2 (Regulation: Activation)    ABL1 || BAX</span></span>
<span id="cb10-10"><a href="step-0-setup-and-defaults.html#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 6 ABL1          BCL2 Trrust2 TRRUST2 (Regulation: Repression)   ABL1 || BCL2</span></span></code></pre></div>
<ul>
<li><ol start="2" style="list-style-type: decimal">
<li>Entrez Mapping Information: To assist you with your gene set, please note that we have tried to build an extensive resource on some of the Entrez IDs for various genes so that you can map many genes to Entrez IDs. Please beware that Excel may distort some gene names like SEPT7 and MARC1 to numeric date values by mistake (we have tried to troubleshoot for some of those situations). It is not exhaustive, so please check additional sources such as NCBI for any genes you cannot find and make adjustments to this default data source, as needed, if you find any discrepancies. And, please email Saniya at <a href="mailto:skhullar2@wisc.edu" class="email">skhullar2@wisc.edu</a>, so she can update the files as needed (thanks for your help!). For many genes that begin with “LOC,” the entrez ID is often the numeric part of the gene name. This mapping will be used as a lookup in the pipeline to help with merging relationships, from disparate sources, reliably!</li>
</ol></li>
</ul>
<p>Please note an example of some rows of our “data//dataDefaults//entrezMappingInfo//infoDFMappingsGeneSymbolAndID.csv”:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="step-0-setup-and-defaults.html#cb11-1" aria-hidden="true" tabindex="-1"></a>someRows <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">200</span><span class="sc">:</span><span class="dv">205</span>)<span class="co"># 10 different rows of the CSV as an example</span></span>
<span id="cb11-2"><a href="step-0-setup-and-defaults.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(infoDF) # 59,307 rows and 2 columns</span></span>
<span id="cb11-3"><a href="step-0-setup-and-defaults.html#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="step-0-setup-and-defaults.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#infoDF[someRows,]  # geneSymbolName mapping to respective entrezIDs. Please double-check.</span></span>
<span id="cb11-5"><a href="step-0-setup-and-defaults.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     geneSymbolName  entrezID</span></span>
<span id="cb11-6"><a href="step-0-setup-and-defaults.html#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 1           15.212      3805</span></span>
<span id="cb11-7"><a href="step-0-setup-and-defaults.html#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 2            47.11      3802</span></span>
<span id="cb11-8"><a href="step-0-setup-and-defaults.html#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 3             3635    201626</span></span>
<span id="cb11-9"><a href="step-0-setup-and-defaults.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 4           Jul-60     11054</span></span>
<span id="cb11-10"><a href="step-0-setup-and-defaults.html#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 5           19-Feb     60343</span></span>
<span id="cb11-11"><a href="step-0-setup-and-defaults.html#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 200        ABHD14B     84836</span></span>
<span id="cb11-12"><a href="step-0-setup-and-defaults.html#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 201         ABHD15    116236</span></span>
<span id="cb11-13"><a href="step-0-setup-and-defaults.html#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 202     ABHD15-AS1 104355133</span></span>
<span id="cb11-14"><a href="step-0-setup-and-defaults.html#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 203        ABHD16A      7920</span></span>
<span id="cb11-15"><a href="step-0-setup-and-defaults.html#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="co"># 204        ABHD16B    140701</span></span>
<span id="cb11-16"><a href="step-0-setup-and-defaults.html#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 205        ABHD17A     81926</span></span></code></pre></div>
<ul>
<li><ol start="3" style="list-style-type: decimal">
<li>Transcription Factor Information: We shared a combined list of Transcription Factors by Lambert and Jasper (as well as EntrezIDs for several of them), and had 1,668 unique TFs.</li>
</ol></li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="step-0-setup-and-defaults.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(tfsDF) # 1,668 rows (TFs) and 2 columns: TFs, entrezID</span></span>
<span id="cb12-2"><a href="step-0-setup-and-defaults.html#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="step-0-setup-and-defaults.html#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(tfsDF) # the 1st 5 rows</span></span>
<span id="cb12-4"><a href="step-0-setup-and-defaults.html#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   TFs entrezID</span></span>
<span id="cb12-5"><a href="step-0-setup-and-defaults.html#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  RUNX1      861</span></span>
<span id="cb12-6"><a href="step-0-setup-and-defaults.html#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 TFAP2A     7020</span></span>
<span id="cb12-7"><a href="step-0-setup-and-defaults.html#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  NR2F1     7025</span></span>
<span id="cb12-8"><a href="step-0-setup-and-defaults.html#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 4  CREB1     1385</span></span>
<span id="cb12-9"><a href="step-0-setup-and-defaults.html#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 5   E2F1     1869</span></span>
<span id="cb12-10"><a href="step-0-setup-and-defaults.html#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 6  NFIL3     4783</span></span></code></pre></div>
<p>To adjust data sources, please note that you can just go to the “data <span class="math inline">\(\rightarrow\)</span> dataDefaults” folder.</p>
<p>Please navigate to code <span class="math inline">\(\rightarrow\)</span> userInputForEachStep and then go to “step0_loadingInDefaultDatasets.R” to make adjustments to any of these parameters below.</p>
<p>Below, please note the main parameters to adjust (and please see step0_loadingInDefaultDatasets.R for additional parameters that may be varied)</p>
<table>
<colgroup>
<col width="33%" />
<col width="33%" />
<col width="33%" />
</colgroup>
<thead>
<tr class="header">
<th>Variable Input Needed by You</th>
<th>Description</th>
<th>Default Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>adsnphenoDirectory</td>
<td>The region of the body that your gene expression data is for. In our example, it is the Hippocampus Ca1 Region</td>
<td>adsnphenoDirectory = “F://organizedAlzheimers//ADSNPheno//”</td>
</tr>
<tr class="even">
<td>tfsUsed</td>
<td>The sources of information for the Transcription Factors (TFs). Here, we use TFs found in Lambert and/or Jaspar as our default.</td>
<td>tfsUsed = “LambertAndJaspar”</td>
</tr>
<tr class="odd">
<td>transcriptionFactorsFilePath</td>
<td>In the dataDefaults &gt; transcriptionFactorInfo directory, this is the name of the CSV file with the list of TFs you want to use. Here, our list is for Lambert and/or Jaspar TFs.</td>
<td>transcriptionFactorsFilePath = “LambertAndJasparTFs.csv”</td>
</tr>
<tr class="even">
<td>geneToEntrezIDMappingPath</td>
<td>In the dataDefaults &gt; entrezMappingInfo directory, this is the name of the CSV file with 2 columns: [Symbols, entrezID]. Here, “Symbols” is the geneName (or # or numeric date), and “entrezID” is what we typically found as the entrezID.</td>
<td>geneToEntrezIDMappingPath = “infoDFMappingsGeneSymbolAndID.csv”</td>
</tr>
<tr class="odd">
<td>additionalGeneRegulatoryNetworkFilePath</td>
<td>In the dataDefaults <span class="math inline">\(\rightarrow\)</span> additionalGeneRegulatoryInfo directory, this is the name of the CSV file with external gold standard gene regulatory network information you would like to include in the final analysis. Here, we used TRRUST2.0. Our data file needs to contain at least these 5 columns: [TF, RegulatedGene, Source, Info, CombinedName]. CombinedName is basically a column based on the gene symbols and is separated with 2 pipes</td>
<td>additionalGeneRegulatoryNetworkFilePath = “Trrust2.csv”</td>
</tr>
<tr class="even">
<td>——</td>
<td>——</td>
<td>——</td>
</tr>
<tr class="odd">
<td>numberOfRoundingDigits</td>
<td>Please note that this is the # of digits after the decimal place for rounding any calculations performed during ADSNPheno.</td>
<td>numberOfRoundingDigits = 3</td>
</tr>
<tr class="even">
<td>includeTimeStamp</td>
<td>Please note that if this Boolean variable is set to TRUE, then additional time-stamped data files will be saved as the pipeline is run.</td>
<td>includeTimeStamp = TRUE</td>
</tr>
<tr class="odd">
<td>disease</td>
<td>Please note that this is where you would mention the disease or condition or relevant biological context. Here, we are analyzing the potential role of non-coding SNPs in Alzheimer’s disease, so we set disease = “Alzheimers.”</td>
<td>disease = “Alzheimers”</td>
</tr>
<tr class="even">
<td>outputPath</td>
<td>Please note that this is where the outputs of ADSNPheno would be stored. By default, ADSNPheno will build an output folder and store the corresponding datafiles and results in sub-folders there.</td>
<td>outputPath = “ADSNPheno//ADSNPhenoOutputs”</td>
</tr>
</tbody>
</table>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-Step0.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/saniyaKhullar/SNPheno/blob/master/02-Step0.Rmd",
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
